var/spack/repos/builtin/packages/py-biom-format/package.py

```diff
@@ -27,4 +32,4 @@ class PyBiomFormat(PythonPackage):
     depends_on('py-scipy@0.13.0:', type=('build', 'run'))
     depends_on('py-pandas@0.19.2:', type=('build', 'run'))
     depends_on('py-six@1.10.0:', type=('build', 'run'))
-    depends_on('py-pyqi', type=('build', 'run'))
+    depends_on('py-pyqi', type=('build', 'run'), when='+pyqi')
```

Comment:
I'm a little confused, the changelog seems to say the pyqi dependency was introduced in 1.2.0 and removed in 2.1.5 and replaced with click.

Reply:
Ah your confusion caused me confusion but I think I've sorted it out now.
Pyqi only works with 3 versions of python (2.7, 3.3, 3.4). This caused conflicts with the installation since other dependencies (namely scipy) required at least 3.6 (of course this depends on which version of biom-format we use). My intention was to limit the call on the Pyqi dependency to only the version of biom-format that needed it. 
For some reason, I thought I read that the transition from pyqi to click happened in 2.6.1 ergo why in the description of the pyqi variant I stated "For use with BIOM 2.1.6". 
Additionally, I think there's problems with the pyqi package too since python-setuptools requires python 3.5.
Let me commit a fix and hopefully, that will clear this up.